python/lsst/dax/obscore/cli/cmd/commands.py

```diff
@@ -30,10 +32,16 @@
     repo_argument,
     where_option,
 )
-from lsst.daf.butler.cli.utils import ButlerCommand, MWPath, split_commas
+from lsst.daf.butler.cli.utils import ButlerCommand, MWPath
 
 from ... import script
 
+# Cast the callback signatures to appease mypy since mypy thinks they
+# are too constrained.
+split_commas = cast(
+    Callable[[click.Context, click.Option | click.Parameter, Any], Any], cmd_utils.split_commas
+)
+
```

Comment:
Does this mean the upstream annotation is wrong?  Or is this just a mess because of the `types-click` vs. types-in-`click` issue?

Reply:
There isn't a types.txt file so it must be picking up the type annotations from the modern click. I'm a bit confused about this one because I thought we tested the annotations for split_commas in other packages.

Reply:
This is actually cargo-culted from daf_butler, I'll try to figure out what is going on.

Reply:
OK, there are couple of issues here:
- `types-click` says that callback second argument needs to be `click.Option | click.Parameter`, while `split_commas` declares it as `click.Option`. `types-click`is probably right, but it does not make sense to me as well because Option is a subclass of Parameter. I think I can change  `split_commas` declaration to `click.Parameter` and it should work fine (and also drop `| None` there).
- `MWOptionDecorator` accepts anything for keyword args (`**kwargs: Any`) so it worked fine with `split_commas`. But for `click.option` decorator that `cast` was used to silence mypy.

I'll try to experiment with callbacks tomorrow to understand all details and fix `split_commas` (and `to_upper`) annotations and run Jenkins to make sure that nothing is broken.

Reply:
Types click should not be used anywhere. Click I comes with its own annotations and the separate package is for older click releases

Reply:
OK, uninstalling types-click silences mypy. Type annotations for `cllick.option` are not super-useful, they are just the same `**kwargs: Any`, but I can live with that. I'll live `daf_butler` alone, will just drop my hack in dax_obscore.